File: AtCoder/Greens/python/ARC017-B-Kaizoudo_ga_Hikui.py
N, K = map(int, input().split())
A = [int(input()) for _ in range(N)]
if K == 1:
    print(N)
    exit()
cnt, res = 0, 0
for i in range(N - 1):
    if A[i + 1] - A[i] > 0:
        cnt += 1
        if cnt + 1 >= K:
            res += 1
    else:
        cnt = 0
print(res)

File: AtCoder/Greens/cpp/ARC017-B-Kaizoudo_ga_Hikui.cpp
#include <bits/stdc++.h>
#define MAX_N 300'000
using namespace std;
typedef long long ll;

ll N, K;
ll A[MAX_N];

int main() {
  cin >> N >> K;
  for (ll i = 0; i < N; ++i) {
    cin >> A[i];
  }
  if (K == 1) {
    printf("%lld\n", N);
    return 0;
  }
  ll cnt = 0;
  ll res = 0;
  for (ll i = 0; i < N - 1; ++i) {
    if (A[i + 1] - A[i] > 0) {
      ++cnt;
      if (cnt + 1 >= K) ++res;
    } else {
      cnt = 0;
    }
  }
  printf("%lld\n", res);
}